### PR TITLE
QOSF Claudia and Georgios changes

### PR DESCRIPTION
**Pull request by clausia**
This PR contains changes made to the Tequila project by two mentees from the **QOSF Mentorship Program Cohort 2**:

**Claudia Zendejas-Morales**:

The functionality of importing and exporting to OpenQASM code and the ability to convert Tequila circuits to PyZX circuits in order to be able to use ZX-calculus.

**Georgios Tsilimigkounakis**:

Creates (or updates) compile_controlled_rotation, compile_controlled_phase, compile_controlled_power.
Removes power_recursor
Removes any code that was "Hadamard" specific. A Hadamard gate is now treated as every other power gate.
Updates compile_to_cc to stop at gates with 1 control. Previously, it was stopping at 2. I believe this increases the compatibility with the backends (or any other external library).
Add supports for compiling a power of Swap gate (+ a simplification for unnecessary controls).
Updates do_compile_trotterized_gate to track the constant term of a generator if it is controlled.
Updates the paramater of PowerGateImpl (multiply by pi) and fixes this in cirq backend.

### CODE REVIEW

File: src/tequila/circuit/_gates_impl.py [hunk 1]
```diff
@@ -268,14 +282,35 @@ def __pow__(self, power, modulo=None):
...
+        to be consitent with exp(-i a/2 G) representation [a: gate.parameter, G: gate.generator]
+    """
+
+    @property
+    def power(self):
+        return self._power
+
+    @power.setter
+    def power(self, other):
+        self._power = assign_variable(variable=other)
```
Comment (kottmanj) [inline src/tequila/circuit/_gates_impl.py:301]: needs to set self.parameter as well to stay consistent

Comment (kottmanj) [reply src/tequila/circuit/_gates_impl.py:301]: see for example dagger operation

**Comment by kottmanj** [comment]
Massive PR. So far it looks really awesome. Great job!!!
Need a bit more time to go through it

File: src/tequila/circuit/_gates_impl.py (hunk 1 above)
Comment (clausia) [reply src/tequila/circuit/_gates_impl.py:301]: @georgios-ts 👆

**Review by kottmanj — APPROVED** [review]

**Pull request merged**